{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ce6d7697",
   "metadata": {},
   "source": [
    "# Annotating the representative genome of *Hymenolepsis microstoma*\n",
    "\n",
    "*Hymenolepsis microstoma* is a small worm. The genome has a size of 169 Mb. The genome was originally published in 2013 ([paper](https://doi.org/10.1038/nature12031)). There is an annotated version of this genome, available under accession number [GCA_000469805.2](https://www.ncbi.nlm.nih.gov/assembly/GCA_000469805.2). There are two problems with this genome:\n",
    "\n",
    "  1. The original annotation is of poor quality (can be measured with BUSCO). \n",
    "  \n",
    "  2. Another genome version - without structural annotation of protein coding genes - was released in the meantime: [GCA_000469805.3](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/469/805/GCA_000469805.3_HMN_v3/).\n",
    "  \n",
    "Our task is today to annotate the new representative genome GCA_000469805.3 with several methods: BRAKER3, BRAKER2, BRAKER1, GALBA, and possibly TSEBRA combinations of the beforementioned, and (a) produce a high quality annotation in a fully automated way, and (b) find out which method produces best results.\n",
    "\n",
    "## Data\n",
    "\n",
    "The genome is available at [/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna](/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna). We will be using the repeat masking provided by NCBI.\n",
    "\n",
    "There are 44 Illumina RNA-Seq libraries listed for this species at the [Sequence Read Archive](https://www.ncbi.nlm.nih.gov/sra). It is not feasible that we download and align all of them. Therefore, we have sampled reads to this genome with VARUS ([paper](https://doi.org/10.1186/s12859-019-3182-x), [software](https://github.com/Gaius-Augustus/VARUS)), provided at [/home/genomics/workshop_material/genome_annotation/sra/Merged.bam](/home/genomics/workshop_material/genome_annotation/sra/Merged.bam).\n",
    "\n",
    "In terms of OrthoDB taxonomy, the Metazoa partition is \"the closest\" for this species, available at [/home/genomics/workshop_material/genome_annotation/orthodb/Metazoa.fa](/home/genomics/workshop_material/genome_annotation/orthodb/Metazoa.fa).\n",
    "\n",
    "Proteomes of close relatives (for GALBA) are [*Opisthokonta*](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/905/250/075/GCA_905250075.1_Rw_10x_megabubbles_genomic_v1/GCA_905250075.1_Rw_10x_megabubbles_genomic_v1_protein.faa.gz), [*Mesocestoides corti*](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/604/375/GCA_900604375.1_M_corti_Specht_Voge_0011_upd/GCA_900604375.1_M_corti_Specht_Voge_0011_upd_protein.faa.gz), [*Rodentolepis nana*](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/617/975/GCA_900617975.1_H_nana_Japan_0011_upd/GCA_900617975.1_H_nana_Japan_0011_upd_protein.faa.gz), [*Echinococcus multilocularis*](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/469/725/GCA_000469725.3_EMULTI002/GCA_000469725.3_EMULTI002_protein.faa.gz), [*Echinococcus granulosus*](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/524/195/GCF_000524195.1_ASM52419v1/GCF_000524195.1_ASM52419v1_protein.faa.gz), and [*Hydatigera taeniaeformis*](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/622/495/GCA_900622495.1_H_taeniaeformis_Canary_Islands_0011_upd/GCA_900622495.1_H_taeniaeformis_Canary_Islands_0011_upd_protein.faa.gz). These protein sets have been downloaded, concatenated, and the fasta entries have been replaced by short, simple, and unique headers. The resulting file to be used with GALBA is available at [/home/genomics/workshop_material/genome_annotation/proteins/proteins.fa](/home/genomics/workshop_material/genome_annotation/proteins/proteins.fa)\n",
    "\n",
    "## Group assignment\n",
    "\n",
    "Runtime is a serious issue. Therefore, you are split into 4 different groups, and you will only complete the task of the group that you were assigned to. Execute the following code cell once to see which group are in:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d96887ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "import random\n",
    "print(\"You are in group \", random.randint(1,5))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11603735",
   "metadata": {},
   "source": [
    "For all tasks, there is a possibility of failure. Should your task fail (e.g. because not sufficient evidence was provided), pick a new group by re-executing the above code cell.\n",
    "\n",
    "## Questions to all groups (with respect to the assigned task)\n",
    "\n",
    "1. After executing the annotation task assigned to you, apply BUSCO to estimate proteome completentess (use [/home/genomics/workshop_material/genome_annotation/metazoa_odb10](/home/genomics/workshop_material/genome_annotation/metazoa_odb10)). What are the BUSCO scores?\n",
    "\n",
    "2. How many genes and how many transcripts are predicted?\n",
    "\n",
    "3. How is the mono:mult ratio? \n",
    "\n",
    "4. What is the median number of exons per transcript? \n",
    "\n",
    "5. What is the largest number of exons in a transcript?\n",
    "\n",
    "## Task for group 1: Apply GALBA\n",
    "\n",
    "Use the following code cell to implement the application of GALBA, and execute GALBA. Use the genome file and the proteins.fa file mentioned, above. Apply `--skipOptimize`. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5dfea1b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%script bash\n",
    "T=4 # adjust to number of threads that you booted with\n",
    "\n",
    "# delete output from a possible previous run if it exists\n",
    "if [ -d GALBA_hymenolepsis ]\n",
    "then\n",
    "    rm -rf GALBA_hymenolepsis\n",
    "fi\n",
    "\n",
    "time galba.pl --workingdir=GALBA_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna \\\n",
    "    --prot_seq=/home/genomics/workshop_material/genome_annotation/proteins/proteins.fa \\\n",
    "    --AUGUSTUS_BIN_PATH=/usr/bin/ --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads ${T} \\\n",
    "    --skipOptimize \\ # remember to remove both options if you are running a real job\n",
    "    2> galba.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d16bccba",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%script bash\n",
    "\n",
    "T=4 # adjust to number of threads that you booted with\n",
    "\n",
    "source conda_init\n",
    "conda activate busco_env\n",
    "\n",
    "cd GALBA_hymenolepsis\n",
    "# create links if not already present\n",
    "if [ ! -d busco_downloads ]\n",
    "then\n",
    "    mkdir busco_downloads\n",
    "    cd busco_downloads\n",
    "    if [ ! -L metazoa_odb10 ]\n",
    "    then\n",
    "        ln -s /home/genomics/workshop_material/genome_annotation/busco/metazoa_odb10 metazoa_odb10\n",
    "    fi\n",
    "fi\n",
    "\n",
    "# delete old output if existing\n",
    "if [ -d busco_${g} ]\n",
    "then\n",
    "    rm -r busco_${g}\n",
    "fi\n",
    "# run BUSCO\n",
    "busco -m proteins -i augustus.hints.aa -o busco_galba_augustus \\\n",
    "        -l metazoa_odb10 -c ${T} &> busco_galba_augustus.log\n",
    "done\n",
    "\n",
    "# Count number of transcripts by counting FASTA headers\n",
    "echo \"Counting number of protein sequences = transcripts\"\n",
    "grep -c \">\" augustus.hints.aa\n",
    "\n",
    "# Number of genes, and descriptive statistics\n",
    "echo \"Computing some descriptive statistics for GALBA:\"\n",
    "analyze_exons.py -f galba.gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b78c11f1",
   "metadata": {},
   "source": [
    "## Task for group 2: Apply BRAKER3\n",
    "\n",
    "Use the following code cell to implement the application of BRAKER3, and execute BRAKER3. Use the genome file, the OrthoDB Metazoa file, the proteins.fa file, and the Merged.bam file mentioned, above. Apply `--skipOptimize`. Do not apply `--gm_max_intergenic 10000`. Expected runtime 10 minutes. If you find the BRAKER gene set of poor quality, consider using TSEBRA, manually."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e76205db",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%script bash\n",
    "\n",
    "T=4 # adjust to number of threads that you booted with, takes ~30 minutes with 4 threads\n",
    "\n",
    "# delete output from a possible previous run if it exists\n",
    "if [ -d BRAKER3_hymenolepsis ]\n",
    "then\n",
    "    rm -rf BRAKER3_hymenolepsis\n",
    "fi\n",
    "\n",
    "ORTHODB=/home/genomics/workshop_material/genome_annotation/orthodb/Metazoa.fa # adjust to suitable clade\n",
    "\n",
    "# run BRAKER3\n",
    "time braker.pl --workingdir=BRAKER3_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna \\\n",
    "    --bam=/home/genomics/workshop_material/genome_annotation/sra/Merged.bam \\\n",
    "    --prot_seq=${ORTHODB},/home/genomics/workshop_material/genome_annotation/proteins/proteins.fa \\\n",
    "    --AUGUSTUS_BIN_PATH=/usr/bin/ \\\n",
    "    --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads ${T} \\\n",
    "    --skipOptimize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c48050e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "63d27777",
   "metadata": {},
   "source": [
    "## Task for group 3: Apply BRAKER2\n",
    "\n",
    "Use the following code cell to implement the application of BRAKER2, and execute BRAKER2. Use the genome file, the OrthoDB Metazoa file and the proteins.fa file mentioned, above. Apply `--skipOptimize`. Do not apply `--gm_max_intergenic 10000`. Expected runtime ... If you find the BRAKER gene set of poor quality, consider using TSEBRA, manually."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94c14622",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%script bash\n",
    "\n",
    "T=4 # adjust to number of threads that you booted with, takes ~30 minutes with 4 threads\n",
    "\n",
    "# delete output from a possible previous run if it exists\n",
    "if [ -d BRAKER2_hymenolepsis ]\n",
    "then\n",
    "    rm -rf BRAKER2_hymenolepsis\n",
    "fi\n",
    "\n",
    "ORTHODB=/home/genomics/workshop_material/genome_annotation/orthodb/Metazoa.fa # adjust to suitable clade\n",
    "\n",
    "# run BRAKER2\n",
    "time braker.pl --workingdir=BRAKER2_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna \\\n",
    "    --prot_seq=${ORTHODB},/home/genomics/workshop_material/genome_annotation/proteins/proteins.fa \\\n",
    "    --AUGUSTUS_BIN_PATH=/usr/bin/ \\\n",
    "    --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads ${T} \\\n",
    "    --skipOptimize"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aeafea05",
   "metadata": {},
   "source": [
    "## Task for group 4: Apply BRAKER1\n",
    "\n",
    "Use the following code cell to implement the application of BRAKER1, and execute BRAKER1. Use the genome file and the Merged.bam file mentioned, above. Apply `--skipOptimize`. Do not apply `--gm_max_intergenic 10000`. Expected runtime ... If you find the BRAKER gene set of poor quality, consider using TSEBRA, manually."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2d2d314d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "#**********************************************************************************\n",
      "#                               BRAKER CONFIGURATION                               \n",
      "#**********************************************************************************\n",
      "# BRAKER CALL: /opt/BRAKER/scripts/braker.pl --workingdir=BRAKER1_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna --bam=/home/genomics/workshop_material/genome_annotation/sra/Merged.bam --AUGUSTUS_BIN_PATH=/usr/bin/ --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads 4 --skipOptimize\n",
      "# Wed May  3 21:50:54 2023: braker.pl version 3.0.3\n",
      "# Wed May  3 21:50:54 2023: Creating directory /home/katharina/git/GenomeAnnotation_Workshop2023/BRAKER1_hymenolepsis.\n",
      "# Wed May  3 21:50:54 2023: Only RNA-Seq input detected, BRAKER will be executed in ET mode (BRAKER1).\n",
      "# Wed May  3 21:50:54 2023: Configuring of BRAKER for using external tools...\n",
      "# Wed May  3 21:50:54 2023: Trying to set $AUGUSTUS_CONFIG_PATH...\n",
      "# Wed May  3 21:50:54 2023: Found environment variable $AUGUSTUS_CONFIG_PATH.\n",
      "# Wed May  3 21:50:54 2023: Checking /usr/share/augustus/config/ as potential path for $AUGUSTUS_CONFIG_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $AUGUSTUS_CONFIG_PATH to /usr/share/augustus/config/!\n",
      "# Wed May  3 21:50:54 2023: WARNING: in file /opt/BRAKER/scripts/braker.pl at line 1894\n",
      "AUGUSTUS_CONFIG_PATH/species (in this case /usr/share/augustus/config//species) is not writeable. BRAKER will try to copy the AUGUSTUS config directory to a writeable location.\n",
      "#**********************************************************************************\n",
      "#                               BRAKER CONFIGURATION                               \n",
      "#**********************************************************************************\n",
      "# BRAKER CALL: /opt/BRAKER/scripts/braker.pl --workingdir=BRAKER1_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna --bam=/home/genomics/workshop_material/genome_annotation/sra/Merged.bam --AUGUSTUS_BIN_PATH=/usr/bin/ --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads 4 --skipOptimize\n",
      "# Wed May  3 21:50:54 2023: braker.pl version 3.0.3\n",
      "# Wed May  3 21:50:54 2023: Creating directory /home/katharina/git/GenomeAnnotation_Workshop2023/BRAKER1_hymenolepsis.\n",
      "# Wed May  3 21:50:54 2023: Only RNA-Seq input detected, BRAKER will be executed in ET mode (BRAKER1).\n",
      "# Wed May  3 21:50:54 2023: Configuring of BRAKER for using external tools...\n",
      "# Wed May  3 21:50:54 2023: Trying to set $AUGUSTUS_CONFIG_PATH...\n",
      "# Wed May  3 21:50:54 2023: Found environment variable $AUGUSTUS_CONFIG_PATH.\n",
      "# Wed May  3 21:50:54 2023: Checking /usr/share/augustus/config/ as potential path for $AUGUSTUS_CONFIG_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $AUGUSTUS_CONFIG_PATH to /usr/share/augustus/config/!\n",
      "# Wed May  3 21:50:54 2023: WARNING: in file /opt/BRAKER/scripts/braker.pl at line 1894\n",
      "AUGUSTUS_CONFIG_PATH/species (in this case /usr/share/augustus/config//species) is not writeable. BRAKER will try to copy the AUGUSTUS config directory to a writeable location.\n",
      "# Wed May  3 21:50:54 2023: Trying to set $AUGUSTUS_BIN_PATH...\n",
      "# Wed May  3 21:50:54 2023: Found command line argument $AUGUSTUS_BIN_PATH.\n",
      "# Wed May  3 21:50:54 2023: Checking /usr/bin as potential path for $AUGUSTUS_BIN_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $AUGUSTUS_BIN_PATH to /usr/bin!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $AUGUSTUS_SCRIPTS_PATH...\n",
      "# Wed May  3 21:50:54 2023: Found command line argument $AUGUSTUS_SCRIPTS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Checking /usr/share/augustus/scripts as potential path for $AUGUSTUS_SCRIPTS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $AUGUSTUS_SCRIPTS_PATH to /usr/share/augustus/scripts!\n",
      "# Wed May  3 21:50:54 2023: WARNING: BRAKER will copy the\n",
      " AUGUSTUS_CONFIG folder into your home directory!\n",
      "# Wed May  3 21:50:54 2023: WARNING: $AUGUSTUS_CONFIG_PATH/species (in this case /usr/share/augustus/config//species ) is not writeable.\n",
      "*** IMPORTANT: Resetting $AUGUSTUS_CONFIG_PATH=/home/katharina/.augustus because BRAKER requires a writable location!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $PYTHON3_PATH...\n",
      "# Wed May  3 21:50:54 2023: Did not find environment variable $PYTHON3_PATH.\n",
      "# Wed May  3 21:50:54 2023: Trying to guess PYTHON3_PATH from location of python3 executable that is available in your $PATH\n",
      "# Wed May  3 21:50:54 2023: Checking /opt/conda/bin as potential path for $PYTHON3_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $PYTHON3_PATH to /opt/conda/bin!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $GENEMARK_PATH...\n",
      "# Wed May  3 21:50:54 2023: Found environment variable $GENEMARK_PATH.\n",
      "# Wed May  3 21:50:54 2023: Checking /opt/ETP/bin as potential path for $GENEMARK_PATH.\n",
      "#*********\n",
      "# WARNING: Couldn't find gmes_petap.pl in /opt/ETP/bin. Will not set $GENEMARK_PATH to /opt/ETP/bin!\n",
      "#*********\n",
      "# Wed May  3 21:50:54 2023: Trying to guess GENEMARK_PATH from location of gmes_petap.pl executable that is available in your $PATH\n",
      "# Wed May  3 21:50:54 2023: Checking /opt/ETP/bin/gmes as potential path for $GENEMARK_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $GENEMARK_PATH to /opt/ETP/bin/gmes!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $BAMTOOLS_PATH...\n",
      "# Wed May  3 21:50:54 2023: Did not find environment variable $BAMTOOLS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Trying to guess BAMTOOLS_PATH from location of bamtools executable that is available in your $PATH\n",
      "# Wed May  3 21:50:54 2023: Checking /usr/bin as potential path for $BAMTOOLS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $BAMTOOLS_PATH to /usr/bin!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $SAMTOOLS_PATH...\n",
      "# Wed May  3 21:50:54 2023: Did not find environment variable $SAMTOOLS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Trying to guess SAMTOOLS_PATH from location of samtools executable that is available in your $PATH\n",
      "# Wed May  3 21:50:54 2023: Checking /usr/bin as potential path for $SAMTOOLS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $SAMTOOLS_PATH to /usr/bin!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $DIAMOND_PATH...\n",
      "# Wed May  3 21:50:54 2023: Did not find environment variable $DIAMOND_PATH.\n",
      "# Wed May  3 21:50:54 2023: Trying to guess DIAMOND_PATH from location of diamond executable that is available in your $PATH\n",
      "# Wed May  3 21:50:54 2023: Checking /opt/ETP/tools as potential path for $DIAMOND_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $DIAMOND_PATH to /opt/ETP/tools!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $TSEBRA_PATH...\n",
      "# Wed May  3 21:50:54 2023: Did not find environment variable $TSEBRA_PATH.\n",
      "# Wed May  3 21:50:54 2023: Trying to guess TSEBRA_PATH from location of tsebra.py executable that is available in your $PATH\n",
      "# Wed May  3 21:50:54 2023: Checking /opt/TSEBRA/bin as potential path for $TSEBRA_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $TSEBRA_PATH to /opt/TSEBRA/bin!\n",
      "# Wed May  3 21:50:54 2023: Trying to set $CDBTOOLS_PATH...\n",
      "# Wed May  3 21:50:54 2023: Did not find environment variable $CDBTOOLS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Trying to guess CDBTOOLS_PATH from location of cdbfasta executable that is available in your $PATH\n",
      "# Wed May  3 21:50:54 2023: Checking /opt/cdbfasta as potential path for $CDBTOOLS_PATH.\n",
      "# Wed May  3 21:50:54 2023: Success! Setting $CDBTOOLS_PATH to /opt/cdbfasta!\n",
      "# Wed May  3 21:50:54 2023: ERROR: in file /opt/BRAKER/scripts/braker.pl at line 3305\n",
      "BAM file /home/genomics/workshop_material/genome_annotation/sra/Merged.bam does not exist.\n",
      "\n",
      "real\t0m0.117s\n",
      "user\t0m0.091s\n",
      "sys\t0m0.017s\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\nT=4 # adjust to number of threads that you booted with, takes ~30 minutes with 4 threads\\n\\n# delete output from a possible previous run if it exists\\nif [ -d BRAKER1_hymenolepsis ]\\nthen\\n    rm -rf BRAKER1_hymenolepsis\\nfi\\n\\n# run BRAKER1\\ntime braker.pl --workingdir=BRAKER1_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna \\\\\\n    --bam=/home/genomics/workshop_material/genome_annotation/sra/Merged.bam \\\\\\n    --AUGUSTUS_BIN_PATH=/usr/bin/ \\\\\\n    --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads ${T} \\\\\\n    --skipOptimize\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[15], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mget_ipython\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun_cell_magic\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mscript\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mbash\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mT=4 # adjust to number of threads that you booted with, takes ~30 minutes with 4 threads\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m# delete output from a possible previous run if it exists\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mif [ -d BRAKER1_hymenolepsis ]\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mthen\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m    rm -rf BRAKER1_hymenolepsis\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mfi\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m# run BRAKER1\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mtime braker.pl --workingdir=BRAKER1_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna \u001b[39;49m\u001b[38;5;130;43;01m\\\\\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m    --bam=/home/genomics/workshop_material/genome_annotation/sra/Merged.bam \u001b[39;49m\u001b[38;5;130;43;01m\\\\\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m    --AUGUSTUS_BIN_PATH=/usr/bin/ \u001b[39;49m\u001b[38;5;130;43;01m\\\\\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m    --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads $\u001b[39;49m\u001b[38;5;132;43;01m{T}\u001b[39;49;00m\u001b[38;5;124;43m \u001b[39;49m\u001b[38;5;130;43;01m\\\\\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m    --skipOptimize\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.10/site-packages/IPython/core/interactiveshell.py:2430\u001b[0m, in \u001b[0;36mInteractiveShell.run_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2428\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuiltin_trap:\n\u001b[1;32m   2429\u001b[0m     args \u001b[38;5;241m=\u001b[39m (magic_arg_s, cell)\n\u001b[0;32m-> 2430\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2432\u001b[0m \u001b[38;5;66;03m# The code below prevents the output from being displayed\u001b[39;00m\n\u001b[1;32m   2433\u001b[0m \u001b[38;5;66;03m# when using magics with decodator @output_can_be_silenced\u001b[39;00m\n\u001b[1;32m   2434\u001b[0m \u001b[38;5;66;03m# when the last Python token in the expression is a ';'.\u001b[39;00m\n\u001b[1;32m   2435\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(fn, magic\u001b[38;5;241m.\u001b[39mMAGIC_OUTPUT_CAN_BE_SILENCED, \u001b[38;5;28;01mFalse\u001b[39;00m):\n",
      "File \u001b[0;32m/opt/conda/lib/python3.10/site-packages/IPython/core/magics/script.py:305\u001b[0m, in \u001b[0;36mScriptMagics.shebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    300\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m args\u001b[38;5;241m.\u001b[39mraise_error \u001b[38;5;129;01mand\u001b[39;00m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    301\u001b[0m     \u001b[38;5;66;03m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[1;32m    302\u001b[0m     \u001b[38;5;66;03m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[1;32m    303\u001b[0m     \u001b[38;5;66;03m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[1;32m    304\u001b[0m     rc \u001b[38;5;241m=\u001b[39m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m9\u001b[39m\n\u001b[0;32m--> 305\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'\\nT=4 # adjust to number of threads that you booted with, takes ~30 minutes with 4 threads\\n\\n# delete output from a possible previous run if it exists\\nif [ -d BRAKER1_hymenolepsis ]\\nthen\\n    rm -rf BRAKER1_hymenolepsis\\nfi\\n\\n# run BRAKER1\\ntime braker.pl --workingdir=BRAKER1_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna \\\\\\n    --bam=/home/genomics/workshop_material/genome_annotation/sra/Merged.bam \\\\\\n    --AUGUSTUS_BIN_PATH=/usr/bin/ \\\\\\n    --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads ${T} \\\\\\n    --skipOptimize\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%script bash\n",
    "\n",
    "T=4 # adjust to number of threads that you booted with, takes ~30 minutes with 4 threads\n",
    "\n",
    "# delete output from a possible previous run if it exists\n",
    "if [ -d BRAKER1_hymenolepsis ]\n",
    "then\n",
    "    rm -rf BRAKER1_hymenolepsis\n",
    "fi\n",
    "\n",
    "# run BRAKER1\n",
    "time braker.pl --workingdir=BRAKER1_hymenolepsis --genome=/home/genomics/workshop_material/genome_annotation/genome/GCA_000469805.3_HMN_v3_genomic.fna \\\n",
    "    --bam=/home/genomics/workshop_material/genome_annotation/sra/Merged.bam \\\n",
    "    --AUGUSTUS_BIN_PATH=/usr/bin/ \\\n",
    "    --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads ${T} \\\n",
    "    --skipOptimize\n",
    "\n",
    "\n",
    "time braker.pl --workingdir=BRAKER1_hymenolepsis --genome=GCA_000469805.3_HMN_v3_genomic.fna \\\n",
    "    --bam=Merged.bam \\\n",
    "    --AUGUSTUS_BIN_PATH=/usr/bin/ \\\n",
    "    --AUGUSTUS_SCRIPTS_PATH=/usr/share/augustus/scripts/ --threads ${T} \\\n",
    "    --skipOptimize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dd1f51a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
